code/modules/defenses/sentry.dm

```diff
@@ -644,13 +658,57 @@
 	QDEL_NULL(linked_cam)
 
 
+/obj/structure/machinery/defenses/sentry/launchable/attackby(obj/item/stack/sheets, mob/user)
+	. = ..()
+
+	if(!istype(sheets, /obj/item/stack/sheet/metal))
+		to_chat(user, SPAN_WARNING("Use [upgrade_cost] metal sheets to give the sentry some plating."))
+		return
+
+	if(upgraded)
+		to_chat(user, SPAN_WARNING("\the [src] has already been upgraded."))
+		return
+
+	if(sheets.amount >= upgrade_cost)
+		if(!do_after(user, 4 SECONDS * user.get_skill_duration_multiplier(SKILL_CONSTRUCTION) , INTERRUPT_ALL, BUSY_ICON_FRIENDLY))
+			to_chat(user, SPAN_WARNING("You were interrupted! Try to stay still while you bolster the sentry with metal sheets..."))
+			return
+
+		src.health_max += health_upgrade
+		src.update_health(-health_upgrade)
+		sheets.use(upgrade_cost)
```

Comment:
if this returns false cancel the upgrade and give an error message (because ppl can remove sheets from the stack during the do_after)